File: task-03/Go/Sub-Task3/code3.go
package main

import (
    "fmt"
    "strings"
)

func printDiamond(n int) {
    // Upper half of the diamond
    for i := 1; i <= n; i++ {
        // Print leading spaces
        fmt.Print(strings.Repeat(" ", n-i))
        // Print asterisks
        fmt.Println(strings.Repeat("*", 2*i-1))
    }

    // Lower half of the diamond
    for i := n - 1; i >= 1; i-- {
        // Print leading spaces
        fmt.Print(strings.Repeat(" ", n-i))
        // Print asterisks
        fmt.Println(strings.Repeat("*", 2*i-1))
    }
}

func main() {
    var n int

    // Prompt user for input
    fmt.Print("Enter the number of rows for the diamond pattern: ")
    _, err := fmt.Scanf("%d", &n)
    if err != nil {
        fmt.Println("Invalid input. Please enter an integer.")
        return
    }

    // Validate input
    if n <= 0 {
        fmt.Println("The number of rows must be greater than 0.")
        return
    }

    // Print the diamond pattern
    printDiamond(n)
}

File: task-03/Elixir/Sub-Task3/code3.ex
defmodule DiamondPattern do
  def print_diamond(n) when n > 0 do
    # Upper half of the diamond
    for i <- 1..n do
      # Print leading spaces
      IO.write(String.duplicate(" ", n - i))
      # Print asterisks
      IO.write(String.duplicate("*", 2 * i - 1))
      IO.puts()
    end

    # Lower half of the diamond
    for i <- n - 1..1 do
      # Print leading spaces
      IO.write(String.duplicate(" ", n - i))
      # Print asterisks
      IO.write(String.duplicate("*", 2 * i - 1))
      IO.puts()
    end
  end

  def main do
    IO.write("Enter the number of rows for the diamond pattern: ")
    n = IO.gets("") |> String.trim() |> String.to_integer()

    if n <= 0 do
      IO.puts("The number of rows must be greater than 0.")
    else
      print_diamond(n)
    end
  end
end

DiamondPattern.main()

